scene/gui/text_edit.cpp

```diff
@@ -3213,12 +3208,11 @@ void TextEdit::set_text_direction(Control::TextDirection p_text_direction) {
 		text.invalidate_font();
 		_update_placeholder();
 
-		if (menu_dir) {
-			menu_dir->set_item_checked(menu_dir->get_item_index(MENU_DIR_INHERITED), text_direction == TEXT_DIRECTION_INHERITED);
-			menu_dir->set_item_checked(menu_dir->get_item_index(MENU_DIR_AUTO), text_direction == TEXT_DIRECTION_AUTO);
-			menu_dir->set_item_checked(menu_dir->get_item_index(MENU_DIR_LTR), text_direction == TEXT_DIRECTION_LTR);
-			menu_dir->set_item_checked(menu_dir->get_item_index(MENU_DIR_RTL), text_direction == TEXT_DIRECTION_RTL);
+		if (!context_menu.is_initialized()) {
+			_generate_context_menu();
```

Comment:
Instead of generating menu, you can skip calling `set_text_direction()` instead.

Reply:
done